### DISCUSSION TITLE
How to calculate PID (2 sources) for 432 parcellations

### DISCUSSION THREAD

**Discussion by max-kat** [pwollstadt/IDTxl#98] [Q&A]
Dear community, 

I am a PhD-student currently trying to implement a PID analysis for one of my experiments. Unfortunately, no one in our institute has ever worked with it and I am coming from a different field and thus would appreciate it if someone could point me in the right direction! Any help is welcome (theoretical & practical)! 

**What I have:**
Experiment with 2 conditions (A&B)
- fMRI data preprocessed, concatonated and parcellated into 432 regions
- results in roughly 800-time points per condition
- discretized continuous data around the mean since the PID calculator apparently only works on discrete data in this package 

**My hypothesis:** 
Based on recent publications and my theory, condition A should lead to higher global synergistic info processing than B 

**What I want to do:**
Calculate an average value for global brain synergistic processing (but for this I probably first need PID for all target and source combinations) 

**Where I am currently stuck:**
I created a list of lists with dimensions [**432**(parcels), **92665**(unique pairwise source combinations, excluding the target and repeated numbers e.g. (2,2)), **2**(the individual source pairs)]
however, reading the manual one has also to supply a list of lags in which the outer list has a length of size target (432) and the inner list has length 2 --> so bivariatePID() does not work since my sources exceed those of the lags.

I have tried a few things but I guess I have a conceptual error... 
Could someone point me in the right direction on how I could use this package to investigate if condition A does lead to higher global synergistic info processing compared to B? 
Also if you do not know exactly how to do something like this but have some theoretical or any other input, I would really be glad to hear it! Anything at this point might help!

Thanks for reading and have a great day!

Best, 
Max

**Comment by mwibral** [comment]
Dear Max,

indeed all publsihed PID estimators so far only work for discretized
data. Please note that your PID results have very strongly depend on
the cosen discretization - potentially with things that looked like
dominated by redudant or unique information  at some resolution looking
completely synergistic at another resolution.

So, to get meaningful results, you must be able to justify the
discretization/quantization of your signals before you dive into PID.
If you have that then you could potentially go ahead. 

For your problem of having soemthing like a 'global' synergisitic
nature of the brain signals, but not a good reason to select certain
voxels as targets, others as sources, it may be better to try the O-
Information measure of Rosas - it may be more to the point with what
you're trying to mesaure.

Best,
Michael

**Reply by max-kat** [reply]
Hi Michael! 

Thanks so much for the quick response! 

Regarding the discretization: 
If I am completely honest (since I am doing this unfortunately more on my own), I took inspiration from a paper by Thomas Varley, who used a similar approach using the HCP dataset. Since the datasets we use are similar and the research question also seems related, I used his work as justification for why I did a discretization around the mean. But I will further look into this, if it is really justified or not.

Ad O-information:
I remember vaguely coming across the O-information, so thank you, I will look further into it and try to make it work!


Do you have any other practical/theoretical tips for a newbee such as me going forward? If not, thank you for your tip with the O-Information!

Have a great day!

Best, 
Max
